google/cloud/firestore/field_path.cc

```diff
@@ -63,10 +61,33 @@ FieldPath FieldPath::Append(FieldPath const& field_path) const {
 }
 
 std::string FieldPath::ToApiRepr() const {
+#if defined(__GNUC__) && __GNUC__ == 4 && __GNUC_MINOR__ == 8
+  // gcc-4.8 ships with a broken regex library (sigh), so don't use it.
+  auto is_simple = [](std::string const& part) {
+    if (part.empty()) {
+      return false;
+    }
+    if (part[0] != '_' and not std::isalpha(part[0])) {
+      return false;
+    }
+    for (auto const& c : part) {
+      if (c != '_' and not std::isalnum(c)) {
+        return false;
+      }
+    }
+    return true;
+  };
+#else
+  auto is_simple = [](std::string const& part) {
```

Comment:
"is_simple_field_name" perhaps?

Reply:
Fixed.  On second thought I removed the code using regex, seems like overkill anyway, happy to restore it if you think it is important.

Reply:
No ... I prefer the non-conditional too.